### ISSUE TITLE
.find() Is Preemptively Truncating id and not Throwing an Error. 

### ISSUE THREAD

**Issue by Megz-Mugz** [rails/rails#54000]
My app has a Model, let's call it model xyz. 

when I do xyz.find(params[:id]) when the params[:id] is something like "26U01010101" it does not throw an error, rather it finds the record with the 26th index. 

This is not what I want it to do, rather it should throw an error if someone tries to access the endpoint /xyz/26U01010101 because "26U01010101" is not a valid index. I can see it is truncating this string and trying to be smart, but this is not ideal.

**Comment by mohsen-alizadeh** [comment]
Could you provide a script to reproduce it?


I am guessing that you try to convert the string to integer. 

For example, you might be running something like `xxx.find("26U01010101".to_i)` 

"26U01010101".to_i returns 26

**Comment by matthewd** [comment]
Yeah, this comes down to the fact we're using `String#to_i`, which has this behaviour.

At this point it's also expected, widely used, and shown in the documentation: https://api.rubyonrails.org/v8.0.1/classes/ActiveRecord/FinderMethods.html#method-i-find
